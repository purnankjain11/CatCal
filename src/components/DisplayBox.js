import React from 'react';
import {View, Text} from 'react-native';

export default class DisplayBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayText : this.props.displayText
    };
  }
  render() {
    return ( <View>
        <Text>{this.state.displayText}</Text>
      </View>);
  }
}