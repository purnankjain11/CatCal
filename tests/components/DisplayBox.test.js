import React from 'react';
import DisplayBox from '../../src/components/DisplayBox';

import renderer from 'react-test-renderer';

describe('component snapshot test' , () => {
  it('renders Display Box without crashing', () => {
    const rendered = renderer.create(<DisplayBox />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});

describe('component prop check', () => {
  it('renders Text passed as prop', () => {
    const displayText = 'Hello';
    const rendered = renderer.create(<DisplayBox displayText={displayText}/>).root;
    expect(rendered.findByType('Text').children[0]).toEqual('Hello');
  });
});
