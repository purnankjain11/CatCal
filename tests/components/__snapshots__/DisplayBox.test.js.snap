// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot test renders Display Box without crashing 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</View>
`;
